mlflow/entities/trace_archive_configuration.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+from typing import Any
+
+from mlflow.entities._mlflow_object import _MlflowObject
+from mlflow.entities.trace_location import TraceLocation
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_trace_server_pb2 import TraceDestination as ProtoTraceDestination
+
+
+@dataclass
+class TraceArchiveConfiguration(_MlflowObject):
```

Comment:
@B-Step62 @jamesbxwu All of these Databricks archival classes need to be marked experimental, and they need to have the `Databricks` in the name. 

Ideally, entities should be vendor-agnostic (e.g. things produced and returned by public, vendor-agnostic MLflow client and fluent APIs), but that's not the case here. `TraceArchiveConfiguration` shouldn't be part of `mlflow.entities`.

I'd prefer something like `mlflow.genai.experimental.enable_databricks_trace_archival(mlflow.genai.experimental.TraceArchiveConfiguration(...)`

Finally, @jamesbxwu can we only expose this API iff the `databricks-agents` SDK is also installed (which is a dependency of `mlflow[databricks]`)? That way, we can be confident that customers are using MLflow on Databricks.